Weird behavior on np.nanmedian

Hello,

I'm no Numba expert, however I have a weird behavior when using nanmedian on arrays containing only nans (edge case).

I'm not sure it was by design, hence this issue.

Thomas

```
import numba as nb
import numpy as np

@nb.jit(nopython=True)
def f(a):
    return np.nansum(a)

a = np.array([np.nan])
print(np.nansum(a))
print(f(a))
```
This is ok!

```
import numba as nb
import numpy as np

@nb.jit(nopython=True)
def f(a):
    return np.nanmedian(a)

a = np.array([np.nan])
print(np.nanmedian(a))
print(f(a))
```
This is not ok:
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-15-cb312ec8285d> in <module>()
      8 a = np.array([np.nan])
      9 print(np.nanmedian(a))
---> 10 print(f(np.array([np.nan, np.nan])))

AssertionError: 
```

--- a/numba/targets/arraymath.py
+++ b/numba/targets/arraymath.py
@@ -770,6 +770,11 @@ if numpy_version >= (1, 9):
                 if not isnan(v):
                     temp_arry[n] = v
                     n += 1
+
+            # all NaNs
+            if n == 0:
+                return np.nan
+
             return _median_inner(temp_arry, n)
 
         return nanmedian_impl

--- a/numba/tests/test_array_reductions.py
+++ b/numba/tests/test_array_reductions.py
@@ -345,6 +345,8 @@ class TestArrayReductions(MemoryLeakMixin, TestCase):
             yield a
             a[a % 4 >= 2] = float('nan')
             yield a
+            a[:] = float('nan')
+            yield a
 
         self.check_median_basic(pyfunc, variations)
 

